guides/ansible_engine/5-role/README.md

```diff
@@ -0,0 +1,221 @@
+
+[source,bash]
+```
+---
+# vars file for apache-simple
+httpd_packages:
+  - httpd
+  - mod_wsgi
+```
+
+---
+**NOTE**
+####
+> Hey, wait just a minute there buster... did you just have us put variables in two seperate places?
+
+Yes... yes we did.  Variables can live in quite a few places.  Just to name a few:
+
+- vars directory
+- defaults directory
+- group_vars directory
+- In the playbook under the `vars:` section
+- In any file which can be specified on the command line using the `--extra_vars` option
+- On a boat, in a moat, with a goat  _(disclaimer:  this is a complete lie)_
+
+Bottom line, you need to read up on [variable precedence](http://docs.ansible.com/ansible/latest/playbooks_variables.html#variable-precedence-where-should-i-put-a-variable) to understand both where to define variables and which locations take precedence.  In this exercise, we are using role defaults to define a couple of variables and these are the most malleable.  After that, we defined some variables in `/vars`
+which have a higher precedence than defaults and can't be overridden as a default variable.
+
+---
+
+### Step 5:
+
+Create your role handler in `roles/apache-simple/handlers/main.yml`.
+
+```yml
+---
+# handlers file for apache-simple
+- name: restart apache service
+  service:
+    name: httpd
+    state: restarted
+    enabled: yes
+```
+
+### Step 6:
+
+Add tasks to your role in `roles/apache-simple/tasks/main.yml`.
+
+```yml
+{% raw %}
+---
+# tasks file for apache-simple
+- name: install httpd packages
+  yum:
+    name: "{{ item }}"
+    state: present
+  with_items: "{{ httpd_packages }}"
+  notify: restart apache service
+
+- name: create site-enabled directory
+  file:
+    name: /etc/httpd/conf/sites-enabled
+    state: directory
+
+- name: copy httpd.conf
+  template:
+    src: templates/httpd.conf.j2
+    dest: /etc/httpd/conf/httpd.conf
+  notify: restart apache service
+
+- name: copy index.html
+  template:
+    src: templates/index.html.j2
+    dest: /var/www/html/index.html
+
+- name: start httpd
+  service:
+    name: httpd
+    state: started
+    enabled: yes
+{% endraw %}    
+```
+
+### Step 7:
+
+Download a couple of templates into `roles/apache-simple/templates/`.  And right after that, let's clean up from exercise 2.1 by removing the old templates directory.
+
+```bash
+mkdir -p ~/apache-basic-playbook/roles/apache-simple/templates/
+cd ~/apache-basic-playbook/roles/apache-simple/templates/
+curl -O http://ansible-workshop.redhatgov.io/workshop-files/httpd.conf.j2
+curl -O http://ansible-workshop.redhatgov.io/workshop-files/index.html.j2
+rm -rf ~/apache-basic-playbook/templates/
+```
+
+## Section 3: Running your new role-based playbook
+
+Now that you've successfully separated your original playbook into a role,
+let's run it and see how it works.
+
+### Step 1:
+
+Run the playbook.
+
+```bash
+ansible-playbook --private-key ~/.ssh/workshopname-tower -i ../hosts site.yml
```

Comment:
Again, please remove `--private-key` and `-i` here.

Reply:
+1.  Private key should not be part of the guide.  Reference - ansible.redhatgov.io